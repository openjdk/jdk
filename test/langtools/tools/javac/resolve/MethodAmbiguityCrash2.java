/*
 * @test /nodynamiccopyright/
 * @bug 8337980
 * @summary Test compiler crash due to failure to resolve method ambiguity
 * @compile/fail/ref=MethodAmbiguityCrash2.out -XDrawDiagnostics MethodAmbiguityCrash2.java
 */
public class MethodAmbiguityCrash2 {

    public interface A {
        int op();
    }

    public abstract static class B {
        public abstract int op();
    }

    public abstract static class C extends B implements A {

        public C(int x) {
        }

        public C() {
            this(op());     // compile should fail here
        }
    }
}
