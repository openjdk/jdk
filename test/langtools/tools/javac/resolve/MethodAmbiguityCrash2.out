MethodAmbiguityCrash2.java:23:18: compiler.err.cant.ref.before.ctor.called: op()
1 error
